sdk/core/core-util/src/bytesEncoding.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+declare global {
+  // stub these out for the browser
+  function btoa(input: string): string;
+  function atob(input: string): string;
+}
+
+import { isNode } from "./checkEnvironment";
+
+export type EncodingType = "utf-8" | "base64" | "base64url";
+
+/**
+ *
+ * @param bytes - the uint8array bytes
+ * @param format - the format we use to encode the byte
+ * @returns a string of the encoded string
+ */
+export function uint8ArrayToString(bytes: Uint8Array, format: EncodingType): string {
+  switch (format) {
+    case "utf-8":
+      return uint8ArrayToUtf8String(bytes);
+    case "base64":
+      return uint8ArrayToBase64(bytes);
+    case "base64url":
+      return uint8ArrayToBase64Url(bytes);
+  }
+}
+
+/**
+ *
+ * @param value - the string to be converted
+ * @param format - the format we use to decode the value
+ * @returns a uint8array
+ */
+export function stringToUint8Array(value: string, format: EncodingType): Uint8Array {
+  switch (format) {
+    case "utf-8":
+      return utf8StringToUint8Array(value);
+    case "base64":
+      return base64ToUint8Array(value);
+    case "base64url":
+      return base64UrlToUint8Array(value);
+  }
+}
+
+/**
+ * Decodes a Uint8Array into a Base64 string.
+ * @internal
+ */
+export function uint8ArrayToBase64(bytes: Uint8Array): string {
+  if (isNode) {
```

Comment:
Instead of doing the check at runtime we should create 2 files one for node and one for browser and set the browser mapping in the package.json

Reply:
You can look at https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-util/src/sha256.browser.ts and https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-util/src/sha256.ts for an example of how to split the implementations.

And this is how you set the browser mapping: https://github.com/Azure/azure-sdk-for-js/blob/8337958585ceabd7edb833678dfe96dcb8c70e6d/sdk/core/core-util/package.json#L9

Reply:
With this, you'd probably be able to get rid of the declare global above as well

Reply:
updated.